assets/js/base/components/payment-methods/saved-payment-method-options.js

```diff
@@ -31,24 +39,46 @@ const getCcOrEcheckPaymentMethodOption = ( { method, expires, tokenId } ) => {
 			expires
 		),
 		name: `wc-saved-payment-method-token-${ tokenId }`,
+		onChange: ( token ) => {
+			const savedTokenKey = `wc-${ method.gateway }-payment-token`;
+			setActivePaymentMethod( method.gateway );
+			setPaymentStatus().success( {
+				payment_method: method.gateway,
+				[ savedTokenKey ]: token,
+			} );
+		},
 	};
 };
 
 /**
  * Returns the option object for any non specific saved payment method.
  *
  * @param {CustomerPaymentMethod} savedPaymentMethod
+ * @param {function(string):void} setActivePaymentMethod
+ * @param {PaymentStatusDispatch} setPaymentStatus
  *
  * @return {Object} An option objects to use for RadioControl.
  */
-const getDefaultPaymentMethodOptions = ( { method, tokenId } ) => {
+const getDefaultPaymentMethodOptions = (
+	{ method, tokenId },
+	setActivePaymentMethod,
+	setPaymentStatus
+) => {
 	return {
 		value: tokenId,
 		label: sprintf(
 			__( 'Saved token for %s', 'woo-gutenberg-products-block' ),
 			method.gateway
 		),
 		name: `wc-saved-payment-method-token-${ tokenId }`,
+		onChange: ( token ) => {
```

Comment:
Feels like this little chunk is a reusable func we could factor out - probably fine as is though. e.g. something like 

```js
function applySavedPaymentMethod( gatewaySlug, token ) { … }

// and then each radio option could use it like
onChange: () => applySavedPaymentMethod( method.gateway, token )
```

Not sure if that's better – just mentioning as it took me a moment to understand.

Reply:
I was wondering if you'd mention that :). It's a good spot, but I usually follow the ["rule of three"](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) before I refactor chunks out.